### PR TITLE
Avoid racing lookups in synchronized context. Put in a const one inst..

### PR DESCRIPTION
**Pull request by baldersheim**
…ead.
@geirst @havardpe PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/features/attributefeature.cpp [hunk 1]
```diff
@@ -277,18 +278,19 @@ AttributeBlueprint::AttributeBlueprint() :
...
 bool
-AttributeBlueprint::setup(const search::fef::IIndexEnvironment & env,
-                          const search::fef::ParameterList & params)
+AttributeBlueprint::setup(const fef::IIndexEnvironment & env,
+                          const fef::ParameterList & params)
 {
     // params[0] = attribute name
     // params[1] = index (array attribute) or key (weighted set attribute)
     _attrName = params[0].getValue();
+    _attrKey = getBaseName() + "." + _attrName;
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/attributefeature.cpp]: if the intent is to share attribute lookups between features, consider using a name explicitly specific for the attribute itself and not for this feature/attribute combination. (NB: beware of name collisions)

File: searchlib/src/vespa/searchlib/features/attributefeature.cpp [hunk 2]
```diff
@@ -413,10 +415,25 @@ createTensorAttributeExecutor(const IAttributeVector *attribute, const vespalib:
 
 }
 
-search::fef::FeatureExecutor &
-AttributeBlueprint::createExecutor(const search::fef::IQueryEnvironment &env, vespalib::Stash &stash) const
+void
+AttributeBlueprint::prepareSharedState(const fef::IQueryEnvironment & env, fef::IObjectStore & store) const
+{
+    if (store.get(_attrKey) == nullptr) {
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/attributefeature.cpp]: if the intent is to NOT share attribute lookups between features, consider removing this 'if' check.

**Review by havardpe — APPROVED** [review]

File: searchlib/src/vespa/searchlib/features/attributefeature.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/attributefeature.cpp]: Intent is to across features. Moved to base class.

File: searchlib/src/vespa/searchlib/features/attributefeature.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/attributefeature.cpp]: Intent to share. Fixed.

**Comment by baldersheim** [comment]
@havardpe Now you can have a look.

**Review by havardpe — COMMENTED** [review]
looks good, but the fef blueprint superclass is probably not the right place for these utility functions

**Comment by baldersheim** [comment]
I put them there as that is common to all blueprints.

**Comment by havardpe** [comment]
I would suggest co-locating the utility code with the object store instead, as a compromise between convenience and decoupling.

**Comment by baldersheim** [comment]
I tried, but it looked even more misplaced...

**Comment by baldersheim** [comment]
I will merge will I ponder on a better place for them.

**Pull request merged**